src/routes/mod.rs

```diff
@@ -171,34 +170,38 @@ pub mod well_known;
 #[response()]
 pub struct CachedFile {
     inner: NamedFile,
-    cache_control: CacheControl,
+    cache_control: Header<'static>,
 }
 
 #[derive(Debug)]
 pub struct ThemeFile(NamedFile);
 
 impl<'r> Responder<'r> for ThemeFile {
-    fn respond_to(self, r: &Request<'_>) -> response::Result<'r> {
-        let contents = std::fs::read(self.0.path()).map_err(|_| Status::InternalServerError)?;
-
-        let mut hasher = DefaultHasher::new();
-        hasher.write(&contents);
-        let etag = format!("{:x}", hasher.finish());
-
-        if r.headers()
-            .get("If-None-Match")
-            .any(|s| s[1..s.len() - 1] == etag)
-        {
-            Response::build()
-                .status(Status::NotModified)
-                .header(ETag(EntityTag::strong(etag)))
-                .ok()
-        } else {
-            Response::build()
-                .merge(self.0.respond_to(r)?)
-                .header(ETag(EntityTag::strong(etag)))
-                .ok()
-        }
+    fn respond_to(self, r: &'r Request<'_>) -> response::ResultFuture<'r> {
+        Box::pin(async move {
+            let contents = std::fs::read(self.0.path()).map_err(|_| Status::InternalServerError)?;
```

Comment:
this std::fs::read could (should?) be async.

Reply:
It could with something like async-std, yes. But maybe add a todo and leave it for a future PR?